### PR TITLE
GitHub: Added stable sideload deployment to the workflow

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

- Deleted stable sideload job from azure pipeline file
- Added stable sideload job to Actions workflow
- Renamed deploy-preview.yml with deployment.yml
- Added some helper ps1 scripts

We are going to have three workflows:
- Files CI (implemented)
- Files Deployment (in progress)
- Files Publication (planning stage)

## PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
- [ ] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   _None_

## Screenshots

_None_

### CODE REVIEW

File: .github/workflows/deploy-preview.yml
Comment (yaira2) [inline .github/workflows/deploy-preview.yml:1]: It might be cleaner to have separate workflows for stable and preview

Comment (0x5bfa) [reply .github/workflows/deploy-preview.yml:1]: Stable and Preview is the same pipeline except having different branch name.

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    - name: Restore Files.sln
+      shell: pwsh
+      run: |
+        msbuild $env:SOLUTION_NAME `
+        -t:Restore `
+        -p:Platform=$env:PLATFORM `
+        -p:Configuration=$env:CONFIGURATION `
+        -p:PublishReadyToRun=true
+
+    - name: Build & package Files
```
Comment (yaira2) [inline .github/workflows/deployment.yml:125]: The store build has a different list of build args

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        -p:Platform=$env:PLATFORM `
+        -p:Configuration=$env:CONFIGURATION `
+        -p:AppxBundlePlatforms=$env:APPX_BUNDLE_PLATFORMS `
+        -p:AppxPackageDir="$env:APPX_PACKAGE_DIR" `
+        -p:AppxBundle=Always `
+        -p:UapAppxPackageBuildMode=Sideload `
+        -p:GenerateAppInstallerFile=True `
+        -p:AppInstallerUri=https://cdn.files.community/files/$env:PACKAGE_DEPLOYMENT_BRANCH_LOWERCASE/
+
+    - name: Sign files with Azure Code Signing
```
Comment (yaira2) [inline .github/workflows/deployment.yml:140]: The store version doesn't need to be signed

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (0x5bfa) [reply .github/workflows/deployment.yml:125]: wdym? This PR is for stable sideload. Store submission still should be ran with Azure Pipeline.

Comment (yaira2) [reply .github/workflows/deployment.yml:125]: I saw some references to the Store build in `Configure-AppxManifest.ps1` but if we're not doing that one yet, you can disregard my comments.

Comment (0x5bfa) [reply .github/workflows/deployment.yml:125]: Yeah it is just preperation.

File: .github/workflows/deployment.yml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+name: Files Deployment
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Select a branch'
+        required: false
+        default: 'Preview'
```
Comment (yaira2) [inline .github/workflows/deployment.yml:16]: Can we force an option to be selected?

Comment (0x5bfa) [reply .github/workflows/deployment.yml:16]: You mean that you'd like to be forced to select at any time?

`required: true` can do it if so.

Comment (yaira2) [reply .github/workflows/deployment.yml:16]: I meant to have nothing selected by default

Comment (0x5bfa) [reply .github/workflows/deployment.yml:16]: You gonna need to select every time. Is that ok?

Comment (yaira2) [reply .github/workflows/deployment.yml:16]: That's what I'm looking for, thanks!

Comment (0x5bfa) [reply .github/workflows/deployment.yml:16]: > Is that ok?

I meant that you are fine with that? You gonna need to select every time. you trigger that pipeline mostly for the preview package. Isn't that gonna be troublesome?

Comment (yaira2) [reply .github/workflows/deployment.yml:16]: Yes

File: .github/workflows/deployment.yml [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  PACKAGE_PROJECT_PATH: 'src\Files.App (Package)\Files.Package.wapproj'
+  TEST_PROJECT_PATH: 'tests\Files.InteractionTests\Files.InteractionTests.csproj'
+  CONFIGURATION: 'Preview' # NOTE: This is just placeholder
+  PLATFORM: 'x64'
+  APPX_BUNDLE_PLATFORMS: 'x64|arm64'
+  WORKING_DIR: ${{ github.workspace }} # Default: D:\a\Files\Files\
+  ARTIFACTS_STAGING_DIR:  ${{ github.workspace }}\artifacts
+  APPX_PACKAGE_DIR: ${{ github.workspace }}\artifacts\AppxPackages
+  PACKAGE_DEPLOYMENT_BRANCH: 'Preview' # NOTE: This is just placeholder
+  PACKAGE_DEPLOYMENT_BRANCH_LOWERCASE: 'preview' # NOTE: This is just placeholder
```
Comment (yaira2) [inline .github/workflows/deployment.yml:34]: The release version needs to use `stable`.

Comment (0x5bfa) [reply .github/workflows/deployment.yml:34]: This is just placeholder as written.
https://github.com/files-community/Files/pull/14218/files#diff-5a17873aec4eae6b52b00959d8f9e17672912858f63181d39de8c3a713e90018R57-R71

**Comment by 0x5bfa** [comment]
I'm gonna open again when Yair decide to switch.

**Comment by yaira2** [comment]
Let's keep open as we're going to need to switch in the next few months

**Comment by 0x5bfa** [comment]
Oh, sure.

**Comment by 0x5bfa** [comment]
I think we should merge soon to test stability.

**Pull request closed without merging**